### ISSUE TITLE
Scanning the same name twice puts all script output under only one of them

### ISSUE THREAD

**Issue by dmiller-nmap** [nmap/nmap#581]
```
./nmap -p80 -Pn --script http-title example.com example.com

Starting Nmap 7.31SVN ( https://nmap.org ) at 2016-11-02 17:08 UTC
Nmap scan report for example.com (93.184.216.34)
Host is up (0.00060s latency).
Other addresses for example.com (not scanned): 2606:2800:220:1:248:1893:25c8:1946
PORT   STATE SERVICE
80/tcp open  http

Nmap scan report for example.com (93.184.216.34)
Host is up (0.00062s latency).
Other addresses for example.com (not scanned): 2606:2800:220:1:248:1893:25c8:1946
PORT   STATE SERVICE
80/tcp open  http
|_http-title: Example Domain
|_http-title: Example Domain

Nmap done: 2 IP addresses (2 hosts up) scanned in 0.39 seconds
```

This is because `host_set_output` and `port_set_output` use `nseU_gettarget` to get the `Target` object for attaching script output, and that only uses the target name or IP to retrieve the `Target`. Since both have the same name and IP, it will always just grab the first one. Two problems to be solved:

1. Keep `Target` objects separate. Maybe add this as a key in the NSE `host` object? Determine if there is any distinction between two `Target` objects specified this way.

2. Assign script output to the proper `Target`. No doubling up and (if there is a distinction) no mixing up of different targets.

**Comment by Varunram** [comment]
> Determine if there is any distinction between two Target objects specified this way.
How exactly do we do go about doing this? And wouldn't it be simpler to just implement a check if two hosts are equal and then assign the script output to them?

**Comment by vinamrabhatia** [comment]
Is the issue still open? Would like to work on it!

**Comment by dmiller-nmap** [comment]
As pointed out in #1219, this also affects virtual hosts, which have different targetnames but the same IP. That issue *should* be solved by addressing this more general one, but it's an important use case to test against.

**Comment by dmiller-nmap** [comment]
Same as #2175, fixed in 4c28124
